content/swiftui/concepts/gestures/gestures.md

```diff
@@ -0,0 +1,137 @@
+  - 'path/iOS-path'
+---
+
+`Views` can respond to taps, drags, pinches and other gestures just by adding **Gesture** modifiers to the view.
+
+## Tap Gestures
+
+This is the most basic type of Gesture.
+The `.onTapGesture()` modifier can be called on the view desired to track the gesture on.
+
+```Swift
+struct Tap: View {
+    var body: some View {
+        Text("Hello, World!")
+            .onTapGesture {
+                print("View Tapped") 
+            }
+    }
+}
+```
+
+The output looks like:
+
+![This is a gif that shows the effect of using TapGesture in Xcode simulation](https://raw.githubusercontent.com/Codecademy/docs/main/media/swiftui-tapgesture-example.gif)
+
+Every time the text view "Hello, World" is tapped, it prints "View Tapped" in the console.
+The action can also be implemented after certain amount of tap gestures. For example:
+
+```Swift
+struct Tap: View {
+    var body: some View {
+        Text("Hello, World!")
+            .onTapGesture(count: 3) {
+                print("View Tapped") 
+            }
+        
+    }
+}
+```
+In this example it prints "View Tapped" after taping the text view three times.
+
+> **Note:** When a button is needed, use a Button instance rather than a tap gesture. Any view can be used as the button’s label, and the button type automatically provides many of the standard behaviors that users expect from a button, like accessibility labels and hints. 
+
+## Long Press Gestures
+
+`onLongPressGesture()` it's used for handling long presses gesture. `onLongPressGesture()` have an extra parameter called `minimumDuration:` (takes Double as input), it's used to specify after how much time the action will be executed. For example:
+
+```Swift
+struct Tap: View {
+    var body: some View {
+        Text("Hello, World!")
+            .onLongPressGesture(minimumDuration: 2) {
+                print("View Tapped")
+            }
+        
+    }
+}
+```
+The output looks like:
+
+![This is a gif that shows the effect of using LongPressGesture in Xcode simulation](https://raw.githubusercontent.com/Codecademy/docs/main/media/swiftui-longgesture-example.gif)
+
+ "Long pressed!" was be printed in the console after pressing the text view for 2 seconds.
+
+## Advanced Gestures
+
+For more advanced and customizable gestures use the `gesture()` modifier with one of the gesture structs like: `DragGesture()`, `LongPressGesture()`, `MagnificationGesture()`, `RotationGesture()`, and `TapGesture()`. These all have special modifiers, usually `onEnded()` and `onChanged() `, and they can be used to take action when the gestures are in-flight (for `onChanged()`) or completed (`for onEnded()`).
+
+As an example, a magnification gesture could be attached to a view so that pinching in and out scales the view up and down. This can be done by creating two `@State` properties to store the scale amount, using that inside a `scaleEffect()` modifier, then setting those values in the gesture, like this:

```

Comment:
```suggestion
In the example below, a magnification gesture could be attached to a view so that pinching in and out scales the view up and down. This can be done by creating two `@State` properties to store the scale amount, using that inside a `.scaleEffect()` modifier, then setting those values in the gesture:

```